### DISCUSSION TITLE
Is there a possibility to add my `allure-cucumber7-junit5-maven` to the official list of `allure-examples`?

### DISCUSSION THREAD

**Discussion by Achitheus** [allure-framework/allure2#2173] [Ideas & Feedback]
When I finished my own project, I decided to update one of the [allure-examples](https://github.com/allure-examples) and ended up with this [allure-cucumber7-junit5-maven](https://github.com/Achitheus/allure-cucumber7-junit5-maven). I'm currently working on creating a report via `GitHub Actions` and would like to add my example to the official [allure-example](https://github.com/allure-examples) list. Is it possible? Unfortunately, most contribute tutorials cover updating/fixing changes, but not creating a new additional repository.
I believe that my example project might help someone a little, and I would be happy to make my first modest contribution to an OS project in my life (I might show the icon to my mom :D )

PS. Is anyone using this [allure-cucumber-jvm](https://mvnrepository.com/artifact/io.qameta.allure/allure-cucumber-jvm)?
I mean this version without a specific number, and not like `allure-cucumberN-jvm`. I ask because this is one of the reasons I spent 3 hard days trying to get my `cucumber7-junit5-maven` project to generate a report correctly (the other reason is of course lack of experience). Because of this, I went the wrong way at first :\

**Comment by baev** [comment]
Hi!

We just launched a [new site](https://allurereport.org) & [new documentation](https://allurereport.org/docs). The section about CucumberJVM is in the review stage and will be published early next week. It will answer all the questions related to the configuration, what cucumber versions are supported, and so on. 

Regarding examples, our team is currently working on an automated solution that will generate example projects for all the supported integrations, releasing ETA mid-November 2023. You may expect a service similar to https://start.spring.io, but for tests.

And what for repositories in the allure-examples organisation: We already have [good coverage for Cucumber-JVM](https://github.com/orgs/allure-examples/repositories?q=cucumber&type=all&language=java), although I haven't yet committed examples for Maven. I'll publish them sometime next week.

For Github Action, I'll suggest using `simple-elf/allure-report-action@v1.7` as in https://github.com/allure-examples/allure-cucumber7-junit-platform-gradle-kts/blob/main/.github/workflows/allure-report.yml#L29-L50 . The action supports history & all the features. To see the generated report, please follow https://allure-examples.github.io/allure-cucumber7-junit-platform-gradle-kts

**Reply by baev** [reply]
If you want to contribute, I may create a repository in allure-examples for you and review the proposed changes ;)

**Reply by Achitheus** [reply]
I must say that the pace of development of the project is impressive. And automatic generation of examples is a cool idea.
As for covering examples, I completely agree if we are talking about Gradle versions. Being a newbie, working only in Maven so far, I still experienced some difficulties.
Special thanks for the advice on GitHub Action. I will study the material and definitely use it.

**Reply by Achitheus** [reply]
> If you want to contribute, I may create a repository in allure-examples for you and review the proposed changes ;)

Yes please. It would make me the happiest mortal! :) Thank you
It's just a pity that I'm a little late. I see that you have everything completely under control.
By the way Initially, I wrote to you personally on LinkedIn, but I didn’t think of writing here right away. 
PS. You have a very cool GitHub profile :) Respect

**Reply by baev** [reply]
Feel free to submit the PR to https://github.com/allure-examples/allure-cucumber7-junit-platform-maven

Use https://github.com/allure-examples/allure-cucumber7-junit-platform-gradle-kts as a baseline, keep the same tests, create a similar readme, GitHub workflow, etc. The only expected change is to replace Gradle with Maven. Also, it's nice to have https://maven.apache.org/wrapper/ configured as well.

**Reply by Achitheus** [reply]
Unfortunately, I already made the PR, but I saw your message here only now. Besides, I’m a little exhausted and would love to study `Maven wrapper` and add support for it later, maybe tomorrow or the day after tomorrow, if you don’t mind :)
In any case, I’m waiting for a reaction to the PR and following its discussion thread

**Comment by Achitheus** [comment]
Summary.
My first ever pull request after adjustments was accepted. After that, I also monitored subsequent commits made after me and did the same for myself.
I’m glad that there was a misunderstanding about `BOMs`. Because of this, I learned a lot about scopes, how `Maven` resolves dependencies, how they are inherited, etc.
I really liked this elegant thing `mvnw`. At first I was dumb and didn’t understand that it allows you to avoid installing dependencies in the workflow. It’s also great that execute permission for file can be given through the `git` command and not clutter up the job with unnecessary steps.
I'll go rework and insert new features into my projects.
Dmitry @baev, thank you for your time and valuable comments!
I'm closing the discussion. Good luck to all!
